pkg/gofr/subscriber.go

```diff
@@ -24,39 +22,58 @@ func newSubscriptionManager(c *container.Container) SubscriptionManager {
 	}
 }
 
-func (s *SubscriptionManager) startSubscriber(topic string, handler SubscribeFunc) {
+func (s *SubscriptionManager) startSubscriber(ctx context.Context, topic string, handler SubscribeFunc) error {
 	// continuously subscribe in an infinite loop
 	for {
-		msg, err := s.container.GetSubscriber().Subscribe(context.Background(), topic)
-		if msg == nil {
-			continue
+		select {
+		case <-ctx.Done():
+			s.container.Logger.Infof("shutting down subscriber for topic %s", topic)
+			return nil
+		default:
+			err := s.handleSubscription(ctx, topic, handler)
+			if err != nil {
+				return err
```

Comment:
Why are we stopping the subscription when there is an error, this is breaking the previous implementation, we tried not to commit the message so that it can be rerereadn the next pass.

Reply:
We can log the error and continue subscribing to the same message!

PS: The error was provided to the handler for the very same purpose, to enable users to commit the message that they consider consumed.